File: CS/NonPersistentDetailViewNavigationExample.Module/DatabaseUpdate/Updater.cs
using System;
using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Updating;
using NonPersistentDetailViewNavigationExample.Module.BusinessObjects;

namespace NonPersistentDetailViewNavigationExample.Module.DatabaseUpdate {
    public class Updater : ModuleUpdater {
        public Updater(IObjectSpace objectSpace, Version currentDBVersion) : base(objectSpace, currentDBVersion) { }
        public override void UpdateDatabaseAfterUpdateSchema() {
            base.UpdateDatabaseAfterUpdateSchema();
#if !EASYTEST
            // Create 100 Issue objects with random Status:
            Random random = new Random();
            for (int i = 0; i < 100; i++) {
                Issue issue = ObjectSpace.CreateObject<Issue>();
                issue.Subject = String.Format("Issue {0:D3}", i);
                if (random.Next(3) == 0) issue.Status = IssueStatus.Active; 
                else issue.Status = IssueStatus.Closed;
            }
#endif
        }
    }
}

File: VB/NonPersistentDetailViewNavigationExample.Module/DatabaseUpdate/Updater.vb
Imports System
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Updating
Imports NonPersistentDetailViewNavigationExample.Module.BusinessObjects

Namespace NonPersistentDetailViewNavigationExample.Module.DatabaseUpdate
    Public Class Updater
        Inherits ModuleUpdater

        Public Sub New(ByVal objectSpace As IObjectSpace, ByVal currentDBVersion As Version)
            MyBase.New(objectSpace, currentDBVersion)
        End Sub
        Public Overrides Sub UpdateDatabaseAfterUpdateSchema()
            MyBase.UpdateDatabaseAfterUpdateSchema()
#If Not EASYTEST Then
            ' Create 100 Issue objects with random Status:
            Dim random As New Random()
            For i As Integer = 0 To 99
                Dim issue As Issue = ObjectSpace.CreateObject(Of Issue)()
                issue.Subject = String.Format("Issue {0:D3}", i)
                If random.Next(3) = 0 Then
                    issue.Status = IssueStatus.Active
                Else
                    issue.Status = IssueStatus.Closed
                End If
            Next i
#End If
        End Sub
    End Class
End Namespace
